Src/IronPython/Compiler/Tokenizer.cs

```diff
@@ -1529,15 +1529,16 @@ private class TokenEqualityComparer : IEqualityComparer<object> {
             }
         }
 
-        internal static bool TryGetEncoding(Encoding defaultEncoding, string line, ref Encoding enc, out string encName) {
-            // PEP 0263 defines the following regex for the encoding line
-            // coding[:=]\s*([-\w.]+)
+        internal static bool TryGetEncoding(string line, ref Encoding enc, out string encName) {
```

Comment:
This method signature is slightly confusing. What about changing it to:
```C#
internal static bool TryGetEncodingName(string line, out string encName);
```
and moving the actual encoding lookup into `GetSourceReader`?
```C#
            // lookup the encoding
            Encoding encoding = defaultEncoding;
            if (gotEncodingName && StringOps.TryGetEncoding(encodingName, out encoding)) {
#if FEATURE_ENCODING
                encoding = (Encoding)encoding.Clone();
                encoding.DecoderFallback = DecoderFallback.ExceptionFallback;
#endif
            }
```

Reply:
I like your suggestion. I too find this method semantics confusing, mostly because of the double meaning of `false` returned from it: it means either that there is no "magic" comment, or there is one but the encoding name is invalid. The way to differentiate them is different handling of the `ref` and `out` arguments.